### PR TITLE
Test target update mapper from discontiguous

### PR DESCRIPTION
**Pull request by nolanbaker31**
Wrote the code, similar to mapper_to_discontiguous. Let me know if you see any issues!

### CODE REVIEW

**Review by spophale — CHANGES_REQUESTED** [review]
Can you delete the empty file "test_target.c" from the commit.

File: tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
+//===--- test_target_update_mapper_from_discontiguous.c -----------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// This test seeks to ensure that target update with motion-clause "from" can properly
+// map data from the device by specifying a user-defined mapper. Additionally, the test
+// checks a new addition to target update in OpenMP 5.0 that states "List items in the
+// to or from clauses may include array sections with stride expressions."
```
Comment (spophale) [inline tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c:8]: Where is the strided expression used ?
The format should be: 
[ lower-bound : length : stride] or
[ lower-bound : : stride] or
[ : length : stride] or
[ : : stride]

**Review by nolanbaker31 — COMMENTED** [review]
Unsure exactly how to use stride expressions & their use, feedback is appreciated

File: tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+typedef struct newvec {
+  size_t len;
+  double *data;
+} newvec_t;
+
+int i;
+int errors = 0;
+
+#pragma omp declare mapper(newvec_t v) map(v, v.data[0:v.len:2])
```
Comment (spophale) [inline tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c:28]: So here you are stating that the array should be mapped with stride of 2 starting at lower bound 0. So 0,2,4,6... indices of the array should be mapped by the mapper.

File: tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  newvec_t s; 
+
+  s.data = (double *)calloc(N,sizeof(double));
+  s.len = N;
+  for (i = 0; i< s.len; i++){ //initalize the struct
+    s.data[i] = 0;
+  }
+  #pragma omp target
+  {
+    for (i = 0; i < s.len; i++) {
```
Comment (spophale) [inline tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c:41]: Here you are accessing all elements of s.data --> the mapper in effect is mapping only even elements. IMO, it should be "i+2".

File: tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      s.data[i] = i;
+    }
+  }//end target
+  #pragma omp target update from(s)
+  for (i =0; i < s.len; i++) {
+    OMPVV_TEST_AND_SET(errors, s.data[i] != i);
```
Comment (spophale) [inline tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c:47]: should test only elements mapped and modified.

**Review by tmh97 — APPROVED** [review]

**Review by spophale — APPROVED** [review]

**Comment by spophale** [comment]
@nolanbaker31 this is ready to merge or do you want to make any more changes ?

**Comment by nolanbaker31** [comment]
@spophale It is ready, but Thomas told me the branch is messed up, so I was about to create a new PR. Do you think it is fine to merge, or should I create a new one?

**Pull request merged**